### PR TITLE
Added force-close command and made manual closing impossible for help-threads

### LINKED ISSUE #80: Reopen thread on user thread close
### Explain the issue.

When a user goes to close a thread manually, reopen the thread and run /close on their behalf. 

### How to recreate?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Zcamt**
Closes #80 

As a result, a new force-close command has been added to allow for the closing of a ticket if needed.

Edit: "manual closing" refers to using the built-in discord button for closing a thread. Instead the close procedure is executed in its place

### CODE REVIEW

File: src/main/kotlin/com/learnspigot/bot/manager/ForumManager.kt [hunk 1]
```diff
@@ -59,6 +61,14 @@ class ForumManager(private val bot: JDA, private val datastore: Datastore, priva
                 forumMessageCounts[it.channel.id] = map
             }
         }
+        bot.listener<ChannelUpdateArchivedEvent> {
+            if(it.newValue != true) return@listener
```
Comment (Nuckerr) [inline src/main/kotlin/com/learnspigot/bot/manager/ForumManager.kt]: !it.newValue

**Review by Nuckerr — CHANGES_REQUESTED** [review]
Are we saving the dontReopen list?

**Comment by Zcamt** [comment]
> Are we saving the dontReopen list?

I don't think we need to.
A channel is only added when the close button is clicked or when force-close I ran. So unless the bot breaks down mid-event, I don't think we need to save it other than just in memory. 

However maybe we should remove the channel from the list afterward, or will that be done by GC?

**Comment by Nuckerr** [comment]
Probably needs removing

**Review by Nuckerr — APPROVED** [review]
Just need to test locally

**Comment by Zcamt** [comment]
> Just need to test locally

Mr Nucker, were you able to test it? :)

**Comment by Nuckerr** [comment]
I have been extremely busy + forgot about this. Shall to tonight

**Comment by Zcamt** [comment]
> I have been extremely busy + forgot about this. Shall to tonight

May I just merge this myself? Or would you still like to test it before?

**Comment by Nuckerr** [comment]
I get this error:
```
17:51:33.818 ForkJoinPool.commonPool-worker-1              RestAction      ERROR  RestAction queue returned failure: [ErrorResponseException] 50083: Thread is archived
net.dv8tion.jda.api.exceptions.ContextException: null
	at net.dv8tion.jda.api.exceptions.ContextException.here(ContextException.java:54)
	at net.dv8tion.jda.api.requests.Request.<init>(Request.java:73)
	at net.dv8tion.jda.internal.requests.RestActionImpl.queue(RestActionImpl.java:200)
	at net.dv8tion.jda.internal.requests.restaction.interactions.InteractionCallbackImpl.queue(InteractionCallbackImpl.java:74)
	at net.dv8tion.jda.api.requests.RestAction.queue(RestAction.java:572)
	at net.dv8tion.jda.api.requests.RestAction.queue(RestAction.java:538)
	at com.learnspigot.bot.command.ThreadCloseCommand$forceThreadCloseCommand$1$1.invokeSuspend(ThreadCloseCommand.kt:53)
	at com.learnspigot.bot.command.ThreadCloseCommand$forceThreadCloseCommand$1$1.invoke(ThreadCloseCommand.kt)
	at com.learnspigot.bot.command.ThreadCloseCommand$forceThreadCloseCommand$1$1.invoke(ThreadCloseCommand.kt)
	at dev.minn.jda.ktx.events.EventsKt$onCommand-5_5nbZA$$inlined$listener-Kx4hsE0$1.onEvent(CoroutineEventManager.kt:153)
	at dev.minn.jda.ktx.events.CoroutineEventManager.runListener$suspendImpl(CoroutineEventManager.kt:101)
	at dev.minn.jda.ktx.events.CoroutineEventManager.runListener(CoroutineEventManager.kt)
	at dev.minn.jda.ktx.events.CoroutineEventManager$handle$1.invokeSuspend(CoroutineEventManager.kt:91)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)

```
and this happens in discord :/
image

**Comment by Zcamt** [comment]
> I get this error:
> 
> ```
> 17:51:33.818 ForkJoinPool.commonPool-worker-1              RestAction      ERROR  RestAction queue returned failure: [ErrorResponseException] 50083: Thread is archived
> net.dv8tion.jda.api.exceptions.ContextException: null
> 	at net.dv8tion.jda.api.exceptions.ContextException.here(ContextException.java:54)
> 	at net.dv8tion.jda.api.requests.Request.<init>(Request.java:73)
> 	at net.dv8tion.jda.internal.requests.RestActionImpl.queue(RestActionImpl.java:200)
> 	at net.dv8tion.jda.internal.requests.restaction.interactions.InteractionCallbackImpl.queue(InteractionCallbackImpl.java:74)
> 	at net.dv8tion.jda.api.requests.RestAction.queue(RestAction.java:572)
> 	at net.dv8tion.jda.api.requests.RestAction.queue(RestAction.java:538)
> 	at com.learnspigot.bot.command.ThreadCloseCommand$forceThreadCloseCommand$1$1.invokeSuspend(ThreadCloseCommand.kt:53)
> 	at com.learnspigot.bot.command.ThreadCloseCommand$forceThreadCloseCommand$1$1.invoke(ThreadCloseCommand.kt)
> 	at com.learnspigot.bot.command.ThreadCloseCommand$forceThreadCloseCommand$1$1.invoke(ThreadCloseCommand.kt)
> 	at dev.minn.jda.ktx.events.EventsKt$onCommand-5_5nbZA$$inlined$listener-Kx4hsE0$1.onEvent(CoroutineEventManager.kt:153)
> 	at dev.minn.jda.ktx.events.CoroutineEventManager.runListener$suspendImpl(CoroutineEventManager.kt:101)
> 	at dev.minn.jda.ktx.events.CoroutineEventManager.runListener(CoroutineEventManager.kt)
> 	at dev.minn.jda.ktx.events.CoroutineEventManager$handle$1.invokeSuspend(CoroutineEventManager.kt:91)
> 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
> 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
> ```
> 
> and this happens in discord :/ image

Could you expand on when this happens? I've tested this countless times and I haven't had this happen at all, are you trying to do this in a thread that's already closed?

**Comment by Zcamt** [comment]
I've now tested the new build I've pushed - everything works as expected and I can't reproduce any error. Please let me know if you end up getting any

**Comment by Nuckerr** [comment]
I will retest this. Also, mind resolving merge conflicts ❤️

**Comment by Zcamt** [comment]
> I will retest this. Also, mind resolving merge conflicts ❤️

Awesome and sure thing, will fix those rn

**Comment by sttephen** [comment]
Great feature. Just going to merge this. 

disaster-girl

**Pull request merged**